doc/doxygen/src/Glossary.dox

```diff
@@ -295,11 +295,12 @@
  *
  * @section manager_state Manager State
  *
- * The @ref ManagerInterface is, by design, a stateless, re-entrant API.
- * Consequently, the result of any call to an instance of this class
- * should solely depend on the methods arguments. In real-world use
- * however, it is often required to understand that a series of discreet
- * calls to the API may be part of the same logical user action.
+ * The @ref ManagerInterface is, by design, a reentrant and thread-safe
```

Comment:
[consider] Just a little wary of saying "thread-safe by design". E.g. given our judicious use of `shared_ptr`, claiming a thread-safe design could raise eyebrows.  Perhaps "thread-safe by contract" or similar?

I wonder if at some point we should support extracting a copy of the underlying object in our `shared_ptr`s, so that manager plugins can be certain that no other thread will mess with the data.

Reply:
Just saw the docs for `ManagerStateBase`, which put it a bit better
```
* The manager interface is by design reentrant, and must be
* thread-safe
```

Reply:
Good point. Tweaked both of those a little [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/1151/commits/aa9af3a604210d2d694a80cddc846e5782338f25)